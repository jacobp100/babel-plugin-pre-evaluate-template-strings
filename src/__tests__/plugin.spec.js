import fs from 'fs';
import { transform } from 'babel-core';

jest.mock('fs');

const plugin = input =>
  transform(input, {
    plugins: [require.resolve('..')],
    babelrc: false,
  }).code;

test('Without tag', () => {
  const code = plugin`
    const world = "world";
    const test = \`
      hello \${world}
    \`
  `;

  expect(code).toMatchSnapshot();
});

test('With tag', () => {
  const code = plugin`
    const world = "world";
    const test = someTag\`
      hello \${world}
    \`
  `;

  expect(code).toMatchSnapshot();
});

test('With external file', () => {
  fs.readFileSync.mockReturnValue(`export default "world";`);

  const code = plugin`
    import world from "/world";
    const test = someTag\`
      hello \${world}
    \`
  `;

  expect(code).toMatchSnapshot();
  expect(fs.readFileSync).toBeCalledWith('/world', 'utf-8');
});

test('With multiple external files', () => {
  fs.readFileSync.mockReturnValueOnce(`
      export { world as default } from "/dictionary";
    `).mockReturnValueOnce(`
      export const world = "world";
    `);

  const code = plugin`
    import world from "/world";

    const test = someTag\`
      hello \${world}
    \`
  `;

  expect(code).toMatchSnapshot();
  expect(fs.readFileSync).toBeCalledWith('/world', 'utf-8');
  expect(fs.readFileSync).toHaveBeenLastCalledWith('/dictionary', 'utf-8');
});
