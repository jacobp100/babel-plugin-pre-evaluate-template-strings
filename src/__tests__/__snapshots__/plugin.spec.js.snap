// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With external duplicate constants file 1`] = `
"
import { test as world } from \\"/world\\";
const test = someTag\`
      hello world
    \`;"
`;

exports[`With external file 1`] = `
"
import world from \\"/world\\";
const test = someTag\`
      hello world
    \`;"
`;

exports[`With external file with duplicate constants 1`] = `
"
import { test as world } from \\"/world\\";
const test = someTag\`
      hello world
    \`;"
`;

exports[`With multiple external files 1`] = `
"
import world from \\"/world\\";

const test = someTag\`
      hello world
    \`;"
`;

exports[`With tag 1`] = `
"
const world = \\"world\\";
const test = someTag\`
      hello world
    \`;"
`;

exports[`Without tag 1`] = `
"
const world = \\"world\\";
const test = \`
      hello world
    \`;"
`;
